### PR TITLE
Add cassava usage example of reading/writing to file

### PR DESCRIPTION
**Pull request by psibi**
I have seen that most (all ?) of my usages with cassava ends up dealing
with writing/reading to a csv file. I think a usage example of that will
be convenient. Also, I suspect this will also be useful for newcomers.

### CODE REVIEW

**Comment by tibbe** [comment]
Sorry for having gotten around to merging this. I actually want an example that recommends that people use the incremental API (i.e. that they don't rely on lazy I/O). I haven't had time to write that example yet.

**Comment by psibi** [comment]
> the incremental API (i.e. that they don't rely on lazy I/O)

Are you talking about the streaming version using pipes/ conduit or just a strict version of ByteString/Text ?

**Comment by tibbe** [comment]
I'm talking about `Data.Csv.Incremental`.

**Comment by psibi** [comment]
I have updated the code to use an incremental API now.

**Comment by codygman** [comment]
@tibbe Inspired by [this thread](https://www.reddit.com/r/haskell/comments/6cbgq3/what_are_the_important_haskell_libs_that_are_no/) I've been perusing the unmerged pull requests on the listed libraries.

This one looks like it could be merged and add value without doing any harm?

File: Data/Csv.hs
Comment (tibbe) [inline Data/Csv.hs]: I've made a habbit of making fields strict in the docs when it makes sense, which it does here.

File: Data/Csv.hs
Comment (tibbe) [inline Data/Csv.hs]: Perhaps this should be in the incremental API module? You could add a forward reference here (e.g. "For an example of achieving the same using the incremental API, see the 'Data.Csv.Incremental' module.")

**Comment by hvr** [comment]
@codygman  please feel free to ping me here or on IRC; I tend to prioritize those projects which people remind me of most often :-)

**Comment by ocramz** [comment]
Ping!

**Comment by andreasabel** [comment]
@psibi @codygman @ocramz  Since this is an old PR, I do not know who is currently behind it.
Would it be possible to fix the conflict and update this PR so that CI can run on it (though CI would not really have to say on this doc-only PR, I figure...)?

**Comment by psibi** [comment]
Unfortunately I don't have spare minutes to pursue this currently. But I don't mind others finishing it to completion.  Free feel to close the PR. Thanks!

**Review by ocramz — APPROVED** [review]
👍

**Comment by ocramz** [comment]
@andreasabel @hvr I don't have write access so cannot merge this

**Pull request merged**
